### ISSUE TITLE
objdetect missing include of ml.hpp

### ISSUE THREAD

**Issue by roman-orekhov** [opencv/opencv#10995]
##### System information (version)

- OpenCV => 3.4.1
- Operating System / Platform => Ubuntu 16.10
- Compiler => Ubuntu 6.2.0-5ubuntu12

##### Detailed description

While trying to compile OpenCV 3.4.1 from sources I get this error:
```.sh
In file included from /ubig/develop/opencv/build/modules/objdetect/opencl_kernels_objdetect.cpp:3:0:
/ubig/develop/opencv/build/modules/objdetect/precomp.hpp:49:26: fatal error: opencv2/ml.hpp: No such file or directory
 #include "opencv2/ml.hpp"
                          ^
```
With verbose output set via cmake-gui I see:
```.sh
cd /ubig/develop/opencv/build/modules/objdetect && /usr/bin/c++   -DCVAPI_EXPORTS -DOPENCV_TRAITS_ENABLE_DEPRECATED -D_USE_MATH_DEFINES -D__OPENCV_BUILD=1 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/ubig/develop/opencv/build/3rdparty/ippicv/ippicv_lnx/include -I/ubig/develop/opencv/build/3rdparty/ippicv/ippiw_lnx/include -I/ubig/develop/opencv/build -isystem /usr/include/eigen3 -I/ubig/develop/opencv/modules/objdetect/include -I/ubig/develop/opencv/modules/objdetect/src -I/ubig/develop/opencv/build/modules/objdetect -I/ubig/develop/opencv/modules/cudev/include -I/ubig/develop/opencv/modules/core/include -I/ubig/develop/opencv/modules/imgproc/include  -fsigned-char -W -Wall -Werror=return-type -Werror=non-virtual-dtor -Werror=address -Werror=sequence-point -Wformat -Werror=format-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Winit-self -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-comment -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffast-math -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG -fPIC   -std=c++11 -o CMakeFiles/opencv_objdetect.dir/opencl_kernels_objdetect.cpp.o -c /ubig/develop/opencv/build/modules/objdetect/opencl_kernels_objdetect.cpp
In file included from /ubig/develop/opencv/build/modules/objdetect/opencl_kernels_objdetect.cpp:3:0:
/ubig/develop/opencv/build/modules/objdetect/precomp.hpp:49:26: fatal error: opencv2/ml.hpp: No such file or directory
 #include "opencv2/ml.hpp"
```

That is, there really isn't any mention of including /ubig/develop/opencv/modules/ml/include/ .
Compiling 3.2 on the same PC didn't have this problem.

##### Steps to reproduce
clone both opencv and opencv contribs, checkout 3.4.1 in both
```.sh
cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/opt/opencv3 -D OPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules -D WITH_CUDA=ON -D CUDA_ARCH_BIN="5.2" -D CUDA_ARCH_PTX="" -D CUDA_HOST_COMPILER=/usr/bin/gcc-5 -D WITH_CUBLAS=ON -D WITH_TBB=ON -D WITH_EIGEN=ON -D ENABLE_FAST_MATH=ON -D CUDA_FAST_MATH=ON -D WITH_LIBV4L=ON -D BUILD_TESTS=OFF -D BUILD_PERF_TESTS=OFF -D BUILD_EXAMPLES=OFF -D WITH_QT=ON -D WITH_OPENGL=ON -D ENABLE_PRECOMPILED_HEADERS=OFF -D BUILD_opencv_stitching=OFF -D BUILD_opencv_tracking=ON -D ENABLE_CXX11=ON ..
make -j 8
```

**Comment by mshabunin** [comment]
@roman-orekhov , please check your OpenCV sources, there is no "ml.hpp" include in the 3.4.1 and the latest versions of this file: 
- _master_ - https://github.com/opencv/opencv/blob/master/modules/objdetect/src/precomp.hpp
- _3.4.1_ - https://github.com/opencv/opencv/blob/3.4.1/modules/objdetect/src/precomp.hpp

**Comment by roman-orekhov** [comment]
The problem appeared to be that I used folder 'build' from previous build. By the way, I didn't see anywhere that one shouldn't use previous build folder for new build, and assumed opencv build system can clean relevant parts by itself. Well, it can't.
Also as soon as I cleared the build dir another problem was gone - python bindings kept complaining about includes of modules which I switched off.
